### PR TITLE
chore: Update individual package Readme files

### PR DESCRIPTION
**Pull request by greghesp**
## Description

Update to the package readme files for `Firestore`, `Functions`, `Auth`, `Core`, `Crashlytics` and `Storage`.

Added usage information to initialize Firebase first, and included a basic usage example

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] If the pull request affects only one plugin, the PR title starts with the name of the plugin in brackets (e.g. [cloud_firestore])
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [ ] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [ ] I read and followed the [Flutter Style Guide].
- [ ] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [ ] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/FirebaseExtended/flutterfire/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://dart.dev/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/firebase_storage/README.md [hunk 1]
```diff
@@ -4,40 +4,42 @@
...
+```dart
+Future<void> uploadFile() {
+  // Create bucket storage reference to not yet existing file
+  Reference ref = FirebaseStorage.instance.ref('/file-upload-test.txt');
 
-// Cancel your subscription when done.
-await uploadTask.onComplete;
-streamSubscription.cancel();
+  // Upload file
+  UploadTask task = ref.putFile(file);
```
Comment (Ehesp) [inline packages/firebase_storage/README.md:36]: Lets add in an example of task usage - listening for events and printing the progress.

Comment (greghesp) [reply packages/firebase_storage/README.md:36]: [Added](https://github.com/FirebaseExtended/flutterfire/pull/3419/commits/43db20f1f4f3ab232a6f5ae76411d80eb6ddf96e)

**Comment by Salakar** [comment]
Added blocked label as this now depends on Functions & Storage re-work landing I'm guessing?

**Comment by Salakar** [comment]
Closing to revisit at a later date.

**Pull request closed without merging**